R/snapshot.R

```diff
@@ -950,37 +950,78 @@ renv_snapshot_report_actions <- function(actions, old, new) {
 
 renv_snapshot_dependencies <- function(project, type = NULL, dev = FALSE) {
 
-  if (!is.null(the$init_dependencies)) {
```

Comment:
This didn't feel right here, since `the$init_dependencies` is populated using `renv_dependencies_impl()`, which doesn't take into account the snapshot type. Maybe we need to make a change on the `init()` side to use `renv_snapshot_dependencies()`?

I think that might be the right change since someone could theoretically do something like:

```
renv::init(settings = list(snapshot.type = "explicit"))
```

but then they'd end up with dependencies from all project files rather than just the DESCRIPTION.

Reply:
That proposal makes sense to me. I think it's important to not recompute here so that the user is only questioned about missing dependencies once when `init`ing